tests/resourcecollection_test.go

```diff
@@ -0,0 +1,499 @@
+				return deployment.GetName() == "deployment1" || deployment.GetNamespace() == collectionNamespaces[0]
+			}),
+		},
+		{
+			"namespace with deployment name",
+			&v1.DryRunCollectionRequest{
+				Name: "test collection",
+				ResourceSelectors: []*storage.ResourceSelector{
+					{
+						Rules: []*storage.SelectorRule{
+							{
+								FieldName: search.Namespace.String(),
+								Operator:  storage.BooleanOperator_OR,
+								Values: []*storage.RuleValue{
+									{
+										Value: collectionNamespaces[0],
+									},
+								},
+							},
+							{
+								FieldName: search.DeploymentName.String(),
+								Operator:  storage.BooleanOperator_OR,
+								Values: []*storage.RuleValue{
+									{
+										Value: "deployment2",
+									},
+								},
+							},
+						},
+					},
+				},
+				EmbeddedCollectionIds: []string{collectionIDs[0]},
+				Options: &v1.CollectionDeploymentMatchOptions{
+					WithMatches: true,
+				},
+			},
+			filter(deploymentList, func(deployment *storage.ListDeployment) bool {
+				return deployment.GetName() == "deployment1" || (deployment.GetNamespace() == collectionNamespaces[0] && deployment.GetName() == "deployment2")
+			}),
+		},
+		{
+			"embedded collection",
+			&v1.DryRunCollectionRequest{
+				Name:                  "test collection",
+				EmbeddedCollectionIds: []string{collectionIDs[1]},
+				Options: &v1.CollectionDeploymentMatchOptions{
+					WithMatches: true,
+				},
+			},
+			filter(deploymentList, func(deployment *storage.ListDeployment) bool {
+				return deployment.GetName() == "deployment2"
+			}),
+		},
+		{
+			"embedded collections",
+			&v1.DryRunCollectionRequest{
+				Name:                  "test collection",
+				EmbeddedCollectionIds: collectionIDs,
+				Options: &v1.CollectionDeploymentMatchOptions{
+					WithMatches: true,
+				},
+			},
+			filter(deploymentList, func(deployment *storage.ListDeployment) bool {
+				return deployment.GetName() == "deployment1" || deployment.GetName() == "deployment2"
+			}),
+		},
+	} {
+		s.T().Run(tc.name, func(t *testing.T) {
+			resp, err := s.service.DryRunCollection(ctx, tc.request)
+			assert.NoError(t, err)
+			assert.ElementsMatch(t, tc.expectedListDeployments, resp.GetDeployments())
+		})
+	}
+
+	// clean up test
+	for _, id := range collectionIDs {
+		_, err = s.service.DeleteCollection(ctx, &v1.ResourceByID{Id: id})
+		assert.NoError(s.T(), err)
+	}
+}
+
+func getTestDeployment(name string, labels ...label) *appsv1.Deployment {
+	ret := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: pointer.Int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "nginx",
```

Comment:
This will pull nginx:latest from docker.io which can cause us to get rate limited. How about `quay.io/rhacs-eng/qa:nginx-1-14-alpine` instead? Or any other image from https://quay.io/repository/rhacs-eng/qa?tab=tags would work as well.